### PR TITLE
Add TS typings

### PR DESCRIPTION
**Pull request by santiagoaguiar**
## Summary

Add typescript typings. They aren't perfect, but they are a start.

This is also a good opportunity to remove anything we _don't_ want to include any longer on the public API...

### CODE REVIEW

**Comment by nw** [comment]
Thank you!

I was going to tackle this in the next release but this is great.

Looks like some synk vulnerabilities popped. Let me get that sorted first.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+
+declare export interface Logger {
```
Comment (hzalaz) [inline index.d.ts]: It 90% accurate the type but bunyan has some variadic parameter args in some cases https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bunyan/index.d.ts#L78 not sure if we should support them or not though

**Comment by santiagoaguiar** [comment]
@nw can we get this one merge? Thanks!!

**Pull request merged**